cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -105,23 +111,25 @@
     .registerTypeAdapter(ScheduleSpecification.class, new ScheduleSpecificationCodec())
     .registerTypeAdapter(WorkflowTokenDetail.class, new WorkflowTokenDetailCodec())
     .registerTypeAdapter(WorkflowTokenNodeDetail.class, new WorkflowTokenNodeDetailCodec())
+    .registerTypeAdapter(Trigger.class, new ProtoTriggerCodec())
+    .registerTypeAdapter(Constraint.class, new ProtoConstraintCodec())
     .create();
 
   private final WorkflowClient workflowClient;
   private final DatasetFramework datasetFramework;
-  private final Scheduler scheduler;
+  private final TimeScheduler timeScheduler;
```

Comment:
So this will change and moved inside the `Scheduler` when the refactoring by @maochf is done, right?